sdk/identity/identity/src/msal/browserFlows/msalImplicit.ts

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as msal from "msal";
+import { MsalBrowserFlowOptions, MsalBrowser } from "./browserCommon";
+import { AccessToken } from "@azure/core-http";
+import { AuthenticationRecord } from "../types";
+import { AuthenticationRequired } from "../errors";
+
+/**
+ * Uses MSAL directly for browser authentication,
+ * which uses the [Implicit Grant Flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-implicit-grant-flow)
+ * @internal
+ */
+export class MSALImplicit extends MsalBrowser {
+  protected app: msal.UserAgentApplication;
+
+  /**
+   * Sets up an MSAL object based on the given parameters.
+   * MSAL with Implicit Grant is not compatible with the account workflow we use from the MSAL Auth Code flow.
+   * In this case, any `authenticationRecord` received will be ignored.
+   * @param options - Parameters necessary and otherwise used to create the MSAL object.
+   */
+  constructor(options: MsalBrowserFlowOptions) {
+    super(options);
+
+    this.msalConfig.cache = {
+      cacheLocation: "localStorage",
+      storeAuthStateInCookie: true // Set to true to improve the experience on IE11 and Edge.
```

Comment:
is there any security tradeoff here?

Reply:
I will ask, but we already shipped code with this.

Reply:
We'll drop implicit Auth after this PR: https://github.com/Azure/azure-sdk-for-js/issues/14276